src/python/pants/util/objects.py

```diff
@@ -2,15 +2,59 @@
 # Licensed under the Apache License, Version 2.0 (see LICENSE).
 
 from abc import ABC, abstractmethod
-from typing import Optional, Type
+from textwrap import dedent
+from typing import Any, Optional, Type, Union
 
 
-# TODO: maybe find a better place for this helper. It probably belongs in `help`, but then we would
-#  have a dependency cycle.
 def get_first_line_of_docstring(cls: Type) -> Optional[str]:
+    """Get the first logical (not necessarily physical) line for a class.
```

Comment:
I don't understand the distinction between a logical and physical line.

Reply:
I'm using the wrong term. By logical line, I mean this:

```python
class FirstSentenceIsMultipleLines:
         """This is all one sentence, it's just really really really long so it stretches to a whole
         new line."""
```

I reread PEP 257 and the better name is `summary line`. I'll rename everything.